File: internal/app/playlists.go
package app

import "github.com/Jota-Music/jota/internal/music"

func (a *App) GetPlaylists() ([]music.PlaylistSummary, error) {
	return a.Playlists.List()
}

func (a *App) CreatePlaylist(name string) (music.PlaylistSummary, error) {
	return a.Playlists.Create(name)
}

func (a *App) DeletePlaylist(id string) error {
	return a.Playlists.Delete(id)
}

func (a *App) AddSongsToPlaylist(id string, refs []string) error {
	return a.Playlists.AddSongs(id, refs)
}

func (a *App) RemoveSongFromPlaylist(id string, ref string) error {
	return a.Playlists.RemoveSong(id, ref)
}

func (a *App) ReorderPlaylist(id string, refs []string) error {
	return a.Playlists.Reorder(id, refs)
}

File: frontend/src/lib/music/app/playlists.ts
import {
	AddSongsToPlaylist,
	CreatePlaylist,
	DeletePlaylist,
	GetPlaylists,
	RemoveSongFromPlaylist,
	ReorderPlaylist,
} from "@bindings/app";
import type { PlaylistSummary } from "@/lib/music/model";

const LOCAL_PREFIX = "local:";

export function isCustom(id: string): boolean {
	return id.startsWith(LOCAL_PREFIX);
}

export async function getPlaylists(): Promise<PlaylistSummary[]> {
	return (await GetPlaylists()) ?? [];
}

export async function createPlaylist(name: string): Promise<PlaylistSummary> {
	return CreatePlaylist(name);
}

export async function deletePlaylist(id: string): Promise<void> {
	await DeletePlaylist(id);
}

export async function addSongsToPlaylist(
	id: string,
	refs: string[],
): Promise<void> {
	await AddSongsToPlaylist(id, refs);
}

export async function removeSongFromPlaylist(
	id: string,
	ref: string,
): Promise<void> {
	await RemoveSongFromPlaylist(id, ref);
}

export async function reorderPlaylist(
	id: string,
	refs: string[],
): Promise<void> {
	await ReorderPlaylist(id, refs);
}
